roles/test-utils/mining-device/src/main.rs

```diff
@@ -286,24 +296,8 @@ impl Device {
 
         let (share_send, share_recv) = async_channel::unbounded();
 
-        let handicap = miner.safe_lock(|m| m.handicap).unwrap();
-        std::thread::spawn(move || loop {
-            std::thread::sleep(std::time::Duration::from_micros(handicap.into()));
-            if miner.safe_lock(|m| m.next_share()).unwrap().is_valid() {
-                let nonce = miner.safe_lock(|m| m.header.unwrap().nonce).unwrap();
-                let time = miner.safe_lock(|m| m.header.unwrap().time).unwrap();
-                let job_id = miner.safe_lock(|m| m.job_id).unwrap();
-                let version = miner.safe_lock(|m| m.version).unwrap();
-                share_send
-                    .try_send((nonce, job_id.unwrap(), version.unwrap(), time))
-                    .unwrap();
-            }
-            miner
-                .safe_lock(|m| m.header.as_mut().map(|h| h.nonce += 1))
-                .unwrap();
-        });
-
-        async_std::task::spawn(async move {
+        start_mining_threads(update_miners, miner, share_send);
```

Comment:
I think it would be possible to combine this call with the call in L300 and the `loop` afterward(all async) in a `tokio::select!` macro, but that could be a different task

Reply:
I mark as solved for now let me know if you still want to use select